### PR TITLE
Wys 2022 upgrades

### PR DESCRIPTION
**Pull request by tahaislam**
## What this pull request accomplishes:

- This branch was created by checking out multiple files from different branches based on their modification dates. The WYS DAG files and helper scripts in the current version of this branch match those currently in `dev_scripts` (except for a minor update in `pull_wys.py` which updates Python path dynamically based on the repo location).
- This branch merges/combines the following branches:
  - wys_api_mobile_fixes
  - wys_open_data_fixes
  - wys_historical_data_fixes
  - wys_empty_json_bug
  - wys_null_fix
- Skipped files from the above branches:
  - `dags/pul_wys_new.py` from `wys_api_mobile_fixes` was excluded from this branch. It adds a new feature to the WYS pipeline that needs to be reviewed and integrated appropriately within the DAG.
  - `wys/api/sql/create_table-mobile_sign_installations_dupes.sql`  from `wys_api_mobile_fixes` was excluded from this branch due to the same reason

## Issue(s) this solves:

Approving and merging this branch should close the aforementioned branches/PRs and their associated issues

### CODE REVIEW

**Comment by radumas** [comment]
We will wait until the monthly dag runs on March 2nd to see how that goes.

**Review by radumas — COMMENTED** [review]
There appears to be a conflict here. 

There's an:
* updated create mat view 
* function to refresh a mat view 

And also a function to inset data into `wys.mobile_api_id`, and I'm pretty sure you can't insert data into mat views.

Ditto for stationary signs.

**Comment by tahaislam** [comment]
> There appears to be a conflict here.
> 
> There's an:
> 
> * updated create mat view
> * function to refresh a mat view
> 
> And also a function to inset data into `wys.mobile_api_id`, and I'm pretty sure you can't insert data into mat views.
> 
> Ditto for stationary signs.

deleted the insertion functions for `wys.mobile_api_id` and `wys.stationary_signs` as these mat views haven't been converted yet into tables

File: wys/api/sql/function-aggregate_speed_counts_one_hour_5kph.sql [hunk 1]
```diff
@@ -16,34 +16,38 @@ BEGIN
...
-	UPDATE wys.raw_data A
-	SET speed_count_uid=B.speed_counts_agg_5kph_id
-	FROM insert_data B
-	WHERE A.speed_count_uid IS NULL
-	AND A.api_id=B.api_id
-	AND A.datetime_bin >= B.datetime_bin AND A.datetime_bin < B.datetime_bin + INTERVAL '1 hour';
+	UPDATE 	wys.raw_data A
+	SET 	speed_count_uid=B.speed_counts_agg_5kph_id
+	FROM 	insert_data B
+	WHERE 	A.speed_count_uid IS NULL AND
```
Comment (radumas) [inline wys/api/sql/function-aggregate_speed_counts_one_hour_5kph.sql:34]: Might speed things up to add `datetime_bin >= _start_date AND datetime_bin < _end_date` here

**Review by radumas — APPROVED** [review]
Looks good. One comment for your consideration.

**Comment by radumas** [comment]
Looks like the automatic linking doesn't happen when the issue number is in the second line of the commit message

**Comment by tahaislam** [comment]
Closed the following PRs:
#404 
#448 
#502 
#507

**Pull request merged**